FirebaseMessaging/Tests/UnitTests/FIRMessagingTokenOperationsTest.m

```diff
@@ -438,4 +440,23 @@ - (void)stubInstallations {
   OCMStub([_mockInstallations authTokenWithCompletion:authTokenWithCompletionArg]);
 }
 
+- (void)setUpHeartbeatStorageForTesting {
+  NSString *const kHeartbeatStorageName = @"HEARTBEAT_INFO_STORAGE";
+  id<GULHeartbeatDateStorable> dataStorage;
```

Comment:
Why is it needed here?

Reply:
Hmm I placed it there because a `dataStorage` is initialized [currently within the test suite](https://github.com/firebase/firebase-ios-sdk/blob/31307cdf7e2e7f0cf931ccf30f54fb409cad037d/FirebaseMessaging/Tests/UnitTests/FIRMessagingTokenOperationsTest.m#L111) but your question makes me wonder why this is needed (because the tests fail when you don't call this  `setUpHeartbeatStorageForTesting`). Investigating...

Reply:
Ah, ok, so for file based storage we need an instance to get the file URL. So I guess it can be moved there. I cannot see why we need to instantiate `GULHeartbeatDateStorageUserDefaults` here (but I may miss something).

Reply:
I understand now: the failing test is written to expect a heartbeat update so we need to clean up the any heartbeat info that may have been generated during a previous test run. Creating a `dataStorage` object is really just a convenient way to remove that data heartbeat data by allowing us to remove the heartbeat file (for non-tvOS) or remove the key in userdefaults (for tvOS).